test/Ocelot.UnitTests/Consul/ConsulServiceDiscoveryProviderTests.cs

```diff
@@ -193,47 +193,18 @@ public void should_not_return_services_with_invalid_port()
                 .And(x => GivenTheServicesAreRegisteredWithConsul(serviceEntryOne, serviceEntryTwo))
                 .When(x => WhenIGetTheServices())
                 .Then(x => ThenTheCountIs(0))
-                .And(x => ThenTheLoggerHasBeenCalledCorrectlyForInvalidPorts())
+                .And(x => ThenTheLoggerHasBeenCalledCorrectlyWithValidationWarning(serviceEntryOne, serviceEntryTwo))
```

Comment:
A-ha-ha! It seems I found the bug made by Tom in https://github.com/ThreeMammals/Ocelot/commit/11a2d13f180a0444d9cb04fd95f87be6489e09d3 on line [61](../commit/11a2d13f180a0444d9cb04fd95f87be6489e09d3#diff-91a3279f52853dbd72d34a3a759a8438e3e83c77cbbde3ef2c526335ee0e98d7R61)...
If address contains `http` **OR** address contains `https` then it is actually **valid** but it returns `false` 🤣 

SO, assumption in the `should_not_return_services_with_invalid_address` test is wrong because `Address = "http://localhost"` is valid, but the test considers it as invalid because of the bug in boolean formula of the predicate. 🥸 
Okay... Tom... Good job! 🤣

Reply:
@ggnaegi Partially fixed in https://github.com/ThreeMammals/Ocelot/pull/1711/commits/5851300815762cfe23304fc23defef2fbbabd119

Regarding `Theory`... You will be able to refactor these 3 tests as you wish in the next PRs... 😉 
Currently I cannot merge 3 tests to one theory because of closure of `_serviceName` private member while `AgentService` creating, but `MemberData` attribute of `Theory` requires only static property/member and it has no access to object members! So, it will be a little challenge for you. 😄

And version you've reviewed just a refactoring, I test the same for which these tests were written. So, port & service name checking is encapsulated inside of the warning expected string.
So, writing something like `actualWarning.ShouldContain(portString)` will be not right.
Finally, I believe, it is not an issue.

Reply:
@raman-m Ok 😄